controller/engine_controller.go

```diff
@@ -1199,24 +1201,37 @@ func restoreBackup(log logrus.FieldLogger, engine *longhorn.Engine, rsMap map[st
 		return errors.Wrapf(err, "cannot get backup target config for backup restoration of engine %v", engine.Name)
 	}
 
+	// Register the name with a backkofentry
+	backoff.Next(engine.Name, time.Now())
 	if cliAPIVersion < engineapi.CLIVersionFour {
 		// For compatible engines, `LastRestoredBackup` is required to indicate if the restore is incremental restore
 		log.Infof("Prepare to restore backup, backup target: %v, backup volume: %v, requested restored backup name: %v, last restored backup name: %v",
 			backupTargetClient.URL, engine.Spec.BackupVolume, engine.Spec.RequestedBackupRestore, engine.Status.LastRestoredBackup)
 		if err = engineClientProxy.BackupRestore(engine, backupTargetClient.URL, engine.Spec.RequestedBackupRestore, engine.Spec.BackupVolume, engine.Status.LastRestoredBackup, backupTargetClient.Credential); err != nil {
+			// only handle the error that is "faield lock"
+			if strings.Contains(err.Error(), "failed lock") && backoff.IsInBackOffSince(engine.Name, time.Now()) {
+				log.Warnf("restoring backup %v failed: %v", engine.Spec.BackupVolume, err)
+				return nil
+			}
```

Comment:
Why not put this error handling inside function `handleRestoreError`?

Reply:
Yes, move it into the function `handleRestoreError`, thanks